lib/ts/hugepages.h

```diff
@@ -20,13 +20,26 @@
  */
 #ifndef _hugepages_h_
 #define _hugepages_h_
+#include "ink_memory.h"
 
 #include <cstring>
 
+#define HUGEPAGE_SIZE  (PAGE_SIZE<<9)
+
 size_t ats_hugepage_size(void);
 bool ats_hugepage_enabled(void);
 void ats_hugepage_init(int);
 void *ats_alloc_hugepage(size_t);
 bool ats_free_hugepage(void *, size_t);
+void *ats_alloc_hugepage_stack(size_t);
+
+void *huge_normal_alloc(void *chunk, size_t size, size_t alignment, bool *zero, bool *commit, unsigned arena_ind);
+void *huge_nodump_alloc(void *chunk, size_t size, size_t alignment, bool *zero, bool *commit, unsigned arena_ind);
+bool huge_dalloc(void *chunk, size_t size, bool committed, unsigned arena_ind);
+bool huge_commit(void *chunk, size_t size, size_t offset, size_t length, unsigned arena_ind);
+bool huge_decommit(void *chunk, size_t size, size_t offset, size_t length, unsigned arena_ind);
+bool huge_purge(void *chunk, size_t size, size_t offset, size_t length, unsigned arena_ind);
+bool huge_split(void *chunk, size_t size, size_t size_a, size_t size_b, bool committed, unsigned arena_ind);
+bool huge_merge(void *chunk_a, size_t size_a, void *chunk_b, size_t size_b, bool committed, unsigned arena_ind);
```

Comment:
There is a different way to do this with the system, however last I check in windows and Linux this required admin/root access. What is being stated here is that this allows a way for use to get a block of memory ( remember we have 64-bit address space which no hardware can use fully) ideally larger than we need. When used for items like arrays, we can simply grow them by having the system move the virtual size of the memory to commit. So a array of 1024 items can be grown to 2048 as we just say the memory we are using it now 2048 items. given the "virtual address" space is free we can do this. The OS maps the virtual address to random hardware for us on the back end, with no extra cost. However I don't think the cost in jemalloc is so low, as it has to manage chucks etc to be handled out. What i don't get is why we are handling these chunks and not jemalloc.